indexer/src/main/java/org/pdxfinder/dataexport/UniversalDataExporter.java

```diff
@@ -0,0 +1,896 @@
+
+    protected DataImportService dataImportService;
+
+    protected String templateDir;
+
+    //data lists
+    private List<List<String>> patientSheetData;
+    private List<List<String>> patientTumorSheetData;
+    private List<List<String>> patientTreatmentSheetData;
+    private List<List<String>> pdxModelSheetData;
+    private List<List<String>> pdxModelValidationSheetData;
+    private List<List<String>> samplePlatformDescriptionSheetData;
+    private List<List<String>> sharingAndContactSheetData;
+    private List<List<String>> cytogeneticsSheetData;
+    private List<List<String>> loaderRelatedDataSheetData;
+    private List<List<String>> drugDosingSheetData;
+
+    private List<List<String>> mutationSheetData;
+    private List<List<String>> cnaSheetData;
+
+    private Group ds;
+
+    private static String notSpecified = "Not Specified";
+    private static String patientOrigin = "patient";
+
+    public UniversalDataExporter() {
+    }
+
+    public UniversalDataExporter(DataImportService dataImportService, UtilityService utilityService) {
+
+        this.dataImportService = dataImportService;
+        this.utilityService = utilityService;
+
+        patientSheetData = new ArrayList<>();
+        patientTumorSheetData = new ArrayList<>();
+        pdxModelSheetData = new ArrayList<>();
+        pdxModelValidationSheetData = new ArrayList<>();
+        sharingAndContactSheetData = new ArrayList<>();
+        loaderRelatedDataSheetData = new ArrayList<>();
+
+        samplePlatformDescriptionSheetData = new ArrayList<>();
+
+        mutationSheetData = new ArrayList<>();
+        cnaSheetData = new ArrayList<>();
+    }
+
+    public Group getDs() {
+        return ds;
+    }
+
+    public void setDs(Group ds) {
+        this.ds = ds;
+    }
+
+    public String getTemplateDir() {
+        return templateDir;
+    }
+
+    public void setTemplateDir(String templateDir) {
+        this.templateDir = templateDir;
+    }
+
+    public void init(String templateDir, Group ds){
+
+        this.templateDir = templateDir;
+        this.ds = ds;
+
+        //:: Methods to initialize the data lists
+
+        initPatientData();
+        initPatientTumorAtCollection();
+        initPdxModelDetails();
+        initPdxModelValidations();
+        initSharingAndContact();
+        initLoaderRelatedData();
+
+        initSamplePlatformDescription();
+
+        initMutationData();
+        initCNAData();
+
+    }
+
+    public void export(String exportDir){
+
+        //get the templates with the headers
+        XSSFWorkbook metadataWorkbook = getWorkbook(templateDir+"/metadata_template.xlsx");
+        XSSFWorkbook samplePlatformWorkbook = getWorkbook(templateDir+"/sampleplatform_template.xlsx");
+        XSSFWorkbook mutationWorkbook = getWorkbook(templateDir+"/mutation_template.xlsx");
+        XSSFWorkbook cnaWorkbook = getWorkbook(templateDir+"/cna_template.xlsx");
+
+
+        //insert data from the datalists to the template
```

Comment:
Probably a good rule of thumb that if you need a comment to describe what a block of code does, that block of code deserves to be factored out to its own function.

Reply:
These comments were added when I designed the method but are not useful anymore. Further factoring out the method is not possible as it would require to have XSSFWorkbook class level fields that would make Spring complain.